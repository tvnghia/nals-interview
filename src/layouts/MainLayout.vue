<template>
  <div>
    <app-navigation />
    <slot />
  </div>
</template>

<script>
import AppNavigation from '@/components/AppNavigation'

export default {
  name: 'MainLayout',

  components: {
    AppNavigation
  }
}
</script>
