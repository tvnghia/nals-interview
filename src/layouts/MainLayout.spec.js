import { shallowMount } from '@vue/test-utils'
import wrapperOps from '@/config-test/wrapper'
import MainLayout from '@/layouts/MainLayout'

const scopedSlots = {
  default: '<p class="default-scoped">Scope slot</p>'
}

Object.assign(wrapperOps, { scopedSlots })

describe('MainLayout.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(MainLayout, wrapperOps)
  })

  afterEach(() => wrapper.vm.$destroy())

  describe('when MainLayout was mounted', () => {
    it('should render correctly', () => {
      expect(wrapper.exists()).toBeTruthy()
      expect(wrapper).toMatchSnapshot()
    })

    it('should render slot', () => {
      expect(wrapper.find('p.default-scoped').exists()).toBeTruthy()
    })
  })
})
