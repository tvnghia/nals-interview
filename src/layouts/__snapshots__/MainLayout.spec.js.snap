// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout.vue when MainLayout was mounted should render correctly 1`] = `
<div>
  <app-navigation-stub></app-navigation-stub>
  <p class="default-scoped">Scope slot</p>
</div>
`;
